File: app/types/fleet.ts
export type TruckStatus =
  | "Idle"
  | "At Plant"
  | "Loading"
  | "En Route"
  | "Arrived On Site"
  | "Pouring"
  | "Post Pour"
  | "Returning"
  | "Washed Out"
  | "Service"
  | string;

export type Truck = {
  id?: number;
  company_id?: string;

  truck_number: string;
  driver_id?: number;
  driver_name?: string;

  latitude?: number;
  longitude?: number;
  speed_mph?: number;

  status: TruckStatus;
  job_number?: string;
  customer_name?: string;
  address?: string;

  device_uuid?: string;
  last_updated?: string;
  created_at?: string;
  updated_at?: string;
};

export type TruckLocation = {
  id?: number;
  company_id?: string;

  truck_number: string;
  driver_id?: number;
  device_uuid?: string;

  latitude: number;
  longitude: number;
  speed_mph?: number;

  status?: TruckStatus;
  job_number?: string;

  recorded_at?: string;
};

export type Driver = {
  id: number;
  company_id?: string;

  name: string;
  pin?: string;
  active: boolean;

  truck_number?: string;
  device_uuid?: string;

  signed_in_at?: string;
  created_at?: string;
  updated_at?: string;
};

export type Device = {
  id?: number;
  company_id?: string;

  device_uuid: string;
  device_name?: string;
  truck_number?: string;

  driver_id?: number;
  driver_name?: string;

  active?: boolean;
  last_seen_at?: string;
};

export type CurrentJob = {
  company_id?: string;

  job_number: string;
  customer_name?: string;
  address?: string;

  ordered_qty?: number;
  delivered_qty?: number;

  status?: string;
};

export type FleetDashboardStats = {
  active_trucks: number;
  active_loads: number;
  en_route: number;
  pouring: number;
  exceptions: number;
};

File: backend/app/schemas/fleet.py
from pydantic import BaseModel
from typing import Optional


class FleetDashboardStats(BaseModel):
    active_trucks: int
    active_loads: int
    en_route: int
    pouring: int
    exceptions: int


class Truck(BaseModel):
    id: int
    company_id: str
    truck_number: str
    driver_id: Optional[int] = None
    driver_name: Optional[str] = None
    latitude: float
    longitude: float
    speed_mph: int
    status: str
    job_number: Optional[str] = None
    customer_name: Optional[str] = None
    address: Optional[str] = None
    device_uuid: Optional[str] = None
    last_updated: str
    created_at: str
    updated_at: str


class Driver(BaseModel):
    id: int
    company_id: str
    name: str
    pin: Optional[str] = None
    active: bool
    truck_number: Optional[str] = None
    device_uuid: Optional[str] = None
    signed_in_at: Optional[str] = None
    created_at: str
    updated_at: str


class Device(BaseModel):
    id: int
    company_id: str
    device_uuid: str
    device_name: str
    truck_number: Optional[str] = None
    driver_id: Optional[int] = None
    driver_name: Optional[str] = None
    active: bool
    last_seen_at: Optional[str] = None